app-drawer/app-drawer.html

```diff
@@ -432,71 +492,51 @@
           dx = -x;
         }
 
-        // Do not fling if there is no difference between the current and end state.
-        if ((isVelocityPositive && dx <= 0) || (!isVelocityPositive && dx >= 0)) {
-          return false;
-        }
-
         // Enforce a minimum transition velocity to make the drawer feel snappy.
         if (isVelocityPositive) {
-          this._velocity = Math.max(this._velocity, this.MIN_TRANSITION_VELOCITY);
+          velocity = Math.max(velocity, this.MIN_TRANSITION_VELOCITY);
+          this.opened = this.position === 'left';
         } else {
-          this._velocity = Math.min(this._velocity, -this.MIN_TRANSITION_VELOCITY);
+          velocity = Math.min(velocity, -this.MIN_TRANSITION_VELOCITY);
+          this.opened = this.position === 'right';
         }
 
-        // Calculate the amount of time needed to finish the transition based on the initial
-        // slope of the timing function.
-        var duration = (this.FLING_INITIAL_SLOPE * dx / this._velocity) + 'ms';
-        this.$.contentContainer.style.transitionDuration = duration;
-        this.$.contentContainer.style.transitionTimingFunction = this.FLING_TIMING_FUNCTION;
-        this.$.scrim.style.transitionDuration = duration;
-        this.$.scrim.style.transitionTimingFunction = this.FLING_TIMING_FUNCTION;
+        // Calculate the amount of time needed to finish the transition based on the
+        // initial slope of the timing function.
+        this._setTransitionDuration((this.FLING_INITIAL_SLOPE * dx / velocity) + 'ms');
+        this._setTransitionTimingFunction(this.FLING_TIMING_FUNCTION);
 
-        // Transform instead of toggling opened to avoid calling Polymer setters.
-        if (isClosingLeft) {
-          this._translateDrawer(-drawerWidth);
-        } else if (isClosingRight) {
-          this._translateDrawer(drawerWidth);
-        } else {
-          this._translateDrawer(0);
-        }
+        this._resetDrawerTranslate();
 
         this._isDrawerFlinging = true;
-        return true;
       },
 
-      _transitionEnd: function(event) {
-        if (Polymer.dom(event).localTarget !== this) {
-          return;
-        }
-
-        this._updateDocScroll();
+      _scrimTransitionend: function(event) {
+        // Focused content should only be reset once - rely scrim opacity because
+        // contentContainer triggers transitionend between translateX(256px) and
+        // translateX(100%), which we don't want since those are the same state.
+        if (event.propertyName === 'opacity') {
```

Comment:
I don't think we need the check here since the only transition that would happen on the scrim is `opacity`.